src/zcash/util.cpp

```diff
@@ -10,3 +11,35 @@ std::vector<unsigned char> convertIntToVectorLE(const uint64_t val_int) {
 
     return bytes;
 }
+
+// Convert bytes into boolean vector.
+std::vector<bool> convertBytesVectorToVector(const std::vector<unsigned char>& bytes) {
+	std::vector<bool> ret;
+	ret.resize(bytes.size() * 8);
+
+    unsigned char c;
+    for (size_t i = 0; i < bytes.size(); i++) {
+        c = bytes.at(i);
+        for (size_t j = 0; j < 8; j++) {
+            ret.at((i*8)+j) = (c >> (7-j)) & 1;
+        }
+    }
+
+    return ret;
+}
+
+// Convert boolean vector to integer
+uint64_t convertVectorToInt(const std::vector<bool>& v) {
+    if (v.size() > 64) {
+        throw std::length_error ("boolean vector can't be larger than 64 bits");
+    }
+
+    uint64_t result = 0;
+    for (size_t i=0; i<v.size();i++) {
+        if (v.at(i)) {
```

Comment:
This is not constant-time; is it used in any contexts where that would matter? Maybe just add a comment on #785 to check this.